lang/structs.go

```diff
@@ -311,10 +311,19 @@ func (obj *StmtRes) Init(data *interfaces.Data) error {
 	if err := obj.Name.Init(data); err != nil {
 		return err
 	}
+	typeMap := make(map[string]struct{})
 	for _, x := range obj.Contents {
 		if err := x.Init(data); err != nil {
 			return err
 		}
+
+		// checks if the current ResField type has already been added to typeMap
+		_, isPresent := typeMap[x.String()]
```

Comment:
I think you have the right idea... Except x.String() is probably not right. You want the field name, not the representation of that whole thing, right? This is trickier because obj.Contents is an interface IIRC... So explain what's going on please =D

Reply:
StmtRes.string() returns the kind of resource as a string (structs.go line 275: `// kind of resource, eg: pkg, file, svc, etc...`), so it seems to me like a logical thing to compare. 

Essentially here, I use a map called `typeMap`. I use a map because of its simple method for checking if an object already exists within it. Here (line 320), putting the current string in typeMap returns a boolean, which I save in `isPresent`, that indicates if we have already seen this type of field. If we have seen it, I return an error, if not, the type is saved into the map to be compared against the next fields to be processed.

I see what you mean that this is kind of an indirect way of testing it, but I can't easily think of a more "correct" way to do it.

Reply:
After looking at this, I still feel that this is valid since StmtRes objects are identified by their `kind` with a string, and that's what I test against to find duplicates.